starlite/middleware/session/base.py

```diff
@@ -0,0 +1,313 @@
+
+
+ONE_DAY_IN_SECONDS = 60 * 60 * 24
+
+
+class BaseBackendConfig(BaseModel):
+    _backend_class: Type["SessionBackend"] = PrivateAttr()
+
+    """Configuration for Session middleware cookies."""
+
+    class Config(BaseConfig):
+        arbitrary_types_allowed = True
+
+    key: constr(min_length=1, max_length=256) = "session"  # type: ignore[valid-type]
+    """
+    Key to use for the cookie inside the header,
+    e.g. `session=<data>` where 'session' is the cookie key and <data> is the session data.
+
+    Notes:
+        - If a session cookie exceeds 4KB in size it is split. In this case the key will be of the format
+            'session-{segment number}'.
+    """
+    max_age: conint(ge=1) = ONE_DAY_IN_SECONDS * 14  # type: ignore[valid-type]
+    """Maximal age of the cookie before its invalidated."""
+    scopes: conlist(Literal["http", "websocket"], min_items=1, max_items=2) = ["http", "websocket"]  # type: ignore[valid-type]
+    """Scopes for the middleware - options are 'http' and 'websocket' with the default being both"""
+    path: str = "/"
+    """Path fragment that must exist in the request url for the cookie to be valid. Defaults to '/'."""
+    domain: Optional[str] = None
+    """Domain for which the cookie is valid."""
+    secure: bool = False
+    """Https is required for the cookie."""
+    httponly: bool = True
+    """Forbids javascript to access the cookie via 'Document.cookie'."""
+    samesite: Literal["lax", "strict", "none"] = "lax"
+    """Controls whether or not a cookie is sent with cross-site requests. Defaults to 'lax'."""
+
+    @property
+    def middleware(self) -> DefineMiddleware:
+        """Use this property to insert the config into a middleware list on one
+        of the application layers.
+
+        Examples:
+
+            ```python
+            from os import urandom
+
+            from starlite import Starlite, Request, get
+            from starlite.middleware.session import SessionCookieConfig
+
+            session_config = SessionCookieConfig(secret=urandom(16))
+
+
+            @get("/")
+            def my_handler(request: Request) -> None:
+                ...
+
+
+            app = Starlite(route_handlers=[my_handler], middleware=[session_config.middleware])
+            ```
+
+        Returns:
+            An instance of DefineMiddleware including 'self' as the config kwarg value.
+        """
+        return DefineMiddleware(SessionMiddleware, backend=self._backend_class(config=self))
+
+
+class ServerSideSessionConfig(BaseBackendConfig):
+    session_id_bytes: int = 32
+    """
+    Number of bytes used to generate a random session-ID
+    """
+
+
+ConfigT = TypeVar("ConfigT", bound=BaseBackendConfig)
+ServerConfigT = TypeVar("ServerConfigT", bound=ServerSideSessionConfig)
+
+
+class SessionBackend(abc.ABC, Generic[ConfigT]):
+    def __init__(self, config: ConfigT) -> None:
+        """Generic session backend, for either client- or server-side
```

Comment:
Since this is what is actually documented in the references - I would suggest you add a bit more content here, e.g.

```text
Abstract SessionBackend. This is the base class that is extended by all other SessionBackends. .....
```

Reply:
Done